core/main/src/firebolt/handlers/voice_guidance_rpc.rs

```diff
@@ -215,9 +209,22 @@ impl VoiceguidanceServer for VoiceguidanceImpl {
         ctx: CallContext,
         request: ListenRequest,
     ) -> RpcResult<ListenerResponse> {
-        // Register for individual change events (no-op if already registered), handlers emit VOICE_GUIDANCE_SETTINGS_CHANGED_EVENT.
-        voice_guidance_settings_enabled_changed(&self.state, &ctx, &request, VOICE_GUIDANCE_CHANGED)
+        if self
```

Comment:
Needed to make sure we're registered for `DeviceEvent::VoiceGuidanceEnabledChanged` events.